client/pages/clubs/[id].tsx

```diff
@@ -1,22 +1,228 @@
-import React from "react";
-import { useRouter } from "next/router";
-import { useAppContext } from "../../context";
+import { useRouter } from 'next/router'
+import React from 'react'
+import styled from 'styled-components'
+import { useAppContext } from '../../context'
+import { desktopFontSize, device, fontWeight, PageTitle } from '../../styles/'
+import { Club } from '../../context/'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faLongArrowAltLeft } from '@fortawesome/free-solid-svg-icons'
+import { BasicEvent } from '../../context/Base'
+import moment from 'moment'
+
+interface BannerProps {
+  backgroundImageUrl: string
+}
+
+const Banner = styled.div`
+  background-image: url(${(props: BannerProps) => props.backgroundImageUrl || ''});
+  background-size: cover;
+  height: 400px;
+`
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin: 0 auto 100px;
+  width: 65%;
+`
 
 export default function ClubDetail() {
-  const { clubs } = useAppContext();
-  const router = useRouter();
+  const { clubs } = useAppContext()
+  const router = useRouter()
+
+  const { id } = router.query
+  if (typeof id !== 'string') return null
+
+  const club = clubs.get(parseInt(id))
+
+  return (
+    <div>
+      <Banner backgroundImageUrl={club.backgroundImageURL} />
+      <Container>
+        <ClubInfo club={club} />
+        <EventsHostedByClub events={club.events} />
+      </Container>
+    </div>
+  )
+}
+
+const IconContainer = styled.div`
+  margin-top: 30px;
+`
+
+const Logo = styled.img`
+  border-radius: 100%;
+  display: block;
+  margin-top: 20px;
+  width: 100px;
+`
+
+const ClubName = styled(PageTitle)`
+  margin-top: 30px;
+`
+
+const Links = styled.div`
+  display: flex;
+  margin-top: 20px;
+`
+
+const SocialLink = styled.div`
+  text-decoration: underline;
+`
+
+const ClubMetaData = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  margin-top: 30px;
+`
+
+const MemberCount = styled.div`
+  font-style: italic;
+  font-weight: ${fontWeight.bold};
+`
+
+const Categories = styled.div`
+  display: flex;
+`
+
+const Category = styled.p`
```

Comment:
multiline text inside a category is probably not desired, should add something to prevent it

Reply:
In that picture there happens to be no multiline text inside `Category` tags, I mean specifically preventing multiline `Category` tags.

All that said, we'll probably end up creating a shared `Tag` component and can defer until then.

P.S. the trick to avoiding multiline is a combination of
```css
  overflow: hidden;
  white-space: nowrap;
```
and setting a fixed `height` for the element